dask/array/reductions.py

```diff
@@ -12,43 +11,36 @@
 from operator import mul
 
 import numpy as np
-from tlz import accumulate, compose, drop, get, partition_all, pluck
+from tlz import accumulate, drop, pluck
 
-from dask import config
+import dask.array as da
 from dask.array import chunk
-from dask.array.blockwise import blockwise
 from dask.array.core import (
     Array,
     _concatenate2,
-    asanyarray,
-    broadcast_to,
     handle_out,
     implements,
     unknown_chunk_message,
 )
 from dask.array.creation import arange, diagonal
 from dask.array.dispatch import divide_lookup, nannumel_lookup, numel_lookup
-from dask.array.numpy_compat import NUMPY_GE_200, ComplexWarning
+from dask.array.numpy_compat import NUMPY_GE_200
 from dask.array.utils import (
     array_safe,
     asarray_safe,
-    compute_meta,
     is_arraylike,
     meta_from_array,
     validate_axis,
 )
 from dask.array.wrap import ones, zeros
 from dask.base import tokenize
-from dask.blockwise import lol_tuples
 from dask.highlevelgraph import HighLevelGraph
-from dask.utils import (
-    apply,
-    deepmap,
-    derived_from,
-    funcname,
-    getargspec,
-    is_series_like,
-)
+from dask.utils import apply, deepmap, derived_from
+
+if da._array_expr_enabled():
```

Comment:
This allows us to keep the actual implementations the same and just switch he imports around. Saves a lot of code duplication